### PR TITLE
Add pypi deployment to Travis-CI pipeline

### LINKED ISSUE #58: Set up rdiff-backup on PyPi
The package rdiff-backup is listed at https://pypi.org/project/rdiff-backup/ but no versions after 0.1.1 have been released and the author anyway is unknown. We should try to get access to this package on PyPi and integrate it to be automatically published from tags released on Github.

### PR DESCRIPTION
**Pull request by ericzolf**
As the subject says.
I've documented the approach in PR #255 but in a nutshell:
- only commits tagged like a version are deployed to PyPI (as there is no draft concept)
- commits tagged like a development version (with .devN at the end) are deployed to test.pypy.org
- the Windows wheel deployment doesn't work yet due (IMHO) to a bug in the Travis scripts

Closes #58 

This PR must absolutely be squashed before merging.

### CODE REVIEW

**Comment by ericzolf** [comment]
Example of deployment is available at https://test.pypi.org/project/rdiff-backup/1.9.0b1.dev8/#files
Don't be surprised, the Windows wheel file is there only because I uploaded it manually with `twine upload --repository-url https://test.pypi.org/legacy/ dist/rdiff_backup-1.9.0b1.dev8-cp37-cp37m-win32.whl`.

File: .travis.yml [hunk 1]
```diff
@@ -13,17 +13,75 @@ python:
...
-  file: dist/*.*
-  skip_cleanup: true
-  draft: true
-  token:
-    secure: V1Ad29+3lIQB1Sxqt8RiNKwferucTIr2mJo1Ka0m8Y1GdscP0ZkvHN4Vzjb0hts+lap8cU24tJ1zq9k22bdsNmoS3/YEg7fwbQSOPCJSYbjwMQeitDc1NP3u1mp5LWiEnQ7cflalIkmWUVpwGbwPSnUQdW3JyheUm4DP6jndGHmv3sC2r4WtZDkVY44SI5OGjvOTStTLnXhVlwXMIBKzw+o9BNdmIcpRvI3UTR+C9mA2yoNE7qgpzALZ4qn7qAmTMJzcVQb22yVZqXVCF8VD72Z66GHx75aatMmsJ/2isOEap19sVYt03fY8lchsTUsdRRCAIInxUvbdfgaEoBWj8Vfv2r/vm5DgWh93amv1RkJdmAYAMTGZG/k20vd7N+sNGKCdAF0W8hj5kriGkNnSQg7dFd+NWrF3pX9XcUkT9IGzv5eT5eYeS+yzgB3YPy8qGKwF5dRkRf7Ylw68j8YOly40rAAfgEx/pcFlYrJh62LDPr5RNhKA679+9UAp0rgkiszNH327bj1i0drqeDIhJAZSXw86B8tJZI1JRBVllKNbwDc5GRFNyms4+pvKCO+lAy5Pcn0bQMUq49qxDsdnM1t2RISivWy5CdkOTNW5WYglhovOSoA8/NjcxmsOpOazCaENEvsB3AogI+LRmpZ7lUZKxjaILuVUM2rwXlKxFB0=
-  on:
-    tags: true
-    all_branches: true
-    condition: $MAKE_STEP == ?dist*
+  - provider: releases
```
Comment (ikus060) [inline .travis.yml]: Just to clarify the role of each block. Would be nice to have a comments about what each deploy block is fow in human word.
e.g.: Push all source, distribution, wheel pacakges for Linux and Windows on github.
Push wheel packages to test.pypi.org when commit is tagged with `*dev`
Push wheel pacakges to pypi.org when commit is tagged.

File: .travis.yml [hunk 2]
```diff
@@ -13,17 +13,75 @@ python:
...
+    user: "__token__"
+    password:
+      secure: QGprJSAZR3X88Ir4qpJpNdel5m6CngEtYlD2DAgwRuyPHClhMhXcayB7NbRqACRdtsqBBdQHSOnPO6jBcrhvTIeFbyhiSWZKME113yn4RFD3E+z/p8otb02Lk0DvFTakeC2qxsJ1nlwxGte7qT5Ac9dL7U5O3SSlWNoq2c3QmoD/d+fVdtjf6tp4QywlOIr4Jyx2WWFBQDSQB2cjY7YfwgcQchXkOoVog83dUID4pTh15HlM1Q9E8k0EKueGrDVkS3hhlbPimv1C1coM/IAIAXaTSR1AZZCZELsVSbBOW1hMaAL4MPpr9ekKx/eErnZiIkCuovBTk0XKvzfOyPluFjnuxazKWH9EU5WutCzfxtXh29fLNLYKaqalgjI024kvzTxVFtxJFvCcnhEjcqgs6sxh4nrzAosK163LViFRlJjgbT+2kVgeWiA51fjFTdIj0ep3/NMN8SgvSWQu7SRA8Vtxx9NLzTu57QPKZdRLOW98XUnnSwWPvFymNJ3JHQtp3tsRvTaLcbECvpI82bQa/9+MEszGXwm5TdM3Ogie7OUgDReq6fYbSCYXli5800QwA8Zj6zwJH2qG7l2VAPm0Gd0O9RPP/U2DCxDVAHPDV8VYzKsLWeCQ0Rsxz0S0d1r1Vss15rZ/C/RaCXUJB9ba4F1uRMh4l8Jm4/eSAcuAryI=
+    distributions: bdist_wheel
+    skip_existing: true
+    skip_cleanup: true
+    on:
+      tags: true
+      all_branches: true
+      condition: $MAKE_STEP =~ ^(bdist_wheel)$ && $TRAVIS_TAG =~ ^v[0-9]+\.[0-9]+\.[0-9]+((a|b|rc)[0-9])?\.dev[0-9]+$
```
Comment (ikus060) [inline .travis.yml]: TRAVIS_TAG
Do you really want to be that restrictive about the value of TRAVIS_TAG ? I mean as long as it end with `.dev[0-9]` it should be fine. I would not restrict the version pattern here.

Comment (ericzolf) [reply .travis.yml]: The problem is that the version tag must respect PEP440 to be accepted by PyPI. I tried if not on purpose :smirk:

**Comment by ottok** [comment]
I looked at this, but I find it a bit difficult to follow, so I won't comment in details. Maybe some inline comments and perhaps extra newlines would make it easier to read.

When you squash the final commit, remember to add "(Closes: #58)" to the git title.

I'll add some comments to #58 itself now about related things.

File: .travis.yml (hunk 1 above)
Comment (ericzolf) [reply .travis.yml]: Done in the last commit.

**Comment by ericzolf** [comment]
@ikus060 you need to accept the changes and close your review, or I can't merge.

**Comment by ikus060** [comment]
Done

File: .travis.yml [hunk 3]
```diff
@@ -122,6 +195,8 @@ jobs:
         - cat ../*.changes
         - mkdir -vp dist
         - cp -v ../*.* dist/
+
+    #--- Build and deploy Debian packages ---
```
Comment (ottok) [inline .travis.yml]: This step is running sdist, not building Debian packages

**Comment by ottok** [comment]
There are a lot of release drafts now, was those maybe an side effect of testing this PR?

image

**Comment by ericzolf** [comment]
I guess, you did it, I saw you playing with an otto_ branch ending with an underscore...  But, yeah, right, it came from the new travis CI config. I'll clean it up, no issue.

**Pull request merged**